### ISSUE TITLE
Handle title text overflow nicely?

### ISSUE THREAD

**Issue by notatestuser** [appetizermonster/hain#131]
A list item's description text uses `text-overflow: ellipsis` to clip overflow nicely when the content is longer than its allotted space. However, titles can clip off long words or wrap to new lines when they are long and contain spaces. This can look a bit funky in some places.

Proposal: Add a new attribute in `SearchResult` to apply `text-overflow: ellipsis`* on the title text. We would only want to do this optionally per module and not for all of them. It's still useful to wrap titles when calculating a long formula in `hain-plugin-math`, for instance.

```js
res.add({
    // ...
    title: longTitle,
    singleLineTitle: true,  // <- applies text-overflow property to the title
    // ...
})
```

Example of long word clip:

![](https://cloud.githubusercontent.com/assets/1255926/14973821/71e2fa48-111f-11e6-93cb-8c49fe3b9c67.png)

Example of multi-line break:

![](https://cloud.githubusercontent.com/assets/1255926/14973923/1d579bbc-1121-11e6-8212-2901caa2d47f.png)

\* The actual styles we need to apply are something like:
```css
text-overflow: ellipsis;
white-space: nowrap;
overflow-x: hidden;
```
I've experimented with this locally and it seems to work fine:

![hainclippedtitle](https://cloud.githubusercontent.com/assets/1255926/14974170/e68e316a-1123-11e6-9507-10f5d7b7ae98.png)

What do you think?

**Comment by appetizermonster** [comment]
Nice proposal, and I think it must be supported, but additional property for `SearchResult` isn't that nice.
I think we can find some better way.

**Comment by notatestuser** [comment]
OK. Do you think it could be at the plugin level?

**Comment by appetizermonster** [comment]
Maybe, but I have no confidence about it.
How do you think about the below one?
```javascript
res.add({
  title: {   // can be String or Object
    text: 'single line text',
    singleLine: true
  },
  ...
});
```

**Comment by notatestuser** [comment]
That looks fine to me.

**Comment by appetizermonster** [comment]
Resolved at 84be79685240d7c8c21cb89e92615abd8dc7f8cc, needs docs

**Comment by notatestuser** [comment]
I had forgot to mention that my only concern is that the object form of the title is no longer backward compatible with previous versions. To work around this we could use a new name for the object form and continue to use a string title for older releases. What do you think?

**Comment by appetizermonster** [comment]
Oh, you don't need to worry about it. I wrote a code to support both string and object. so it has backward compatibility. no need to workaround.

**Comment by notatestuser** [comment]
I see. And a new plugin in an older hain will not produce problems either?

**Comment by appetizermonster** [comment]
No, Sorry, I misunderstood. new plugin with new API will not be listed in older hain if they write new API version in `package.json` or they can handle it by using `appContext.CURRENT_API_VERSION`.

**Comment by notatestuser** [comment]
That's great - thanks.

**Comment by notatestuser** [comment]
The titles appear to hit the `text-util` as strings of `[object Object]`. Is react stringifying them?

**Comment by notatestuser** [comment]
Bump @appetizermonster :wink:

**Comment by appetizermonster** [comment]
@notatestuser Oh sorry, I'll make a patch for it soon.

**Comment by notatestuser** [comment]
Thanks!

Get Outlook for iOS




On Thu, Jun 16, 2016 at 10:53 AM +0800, "Heejin Lee" <notifications@github.com> wrote:












@notatestuser Oh sorry, I'll make a patch for it soon.

**Comment by appetizermonster** [comment]
Resolved at 202287c652dba3bfd2853a4a1a83b5a2af374d56

**Comment by notatestuser** [comment]
Great! Please bump up the API version in the next release so I can target this fix. Thanks. :smile:

**Comment by appetizermonster** [comment]
Okay, I'll

**Comment by notatestuser** [comment]
Hey, do you have any idea when this will be released? :smiley:

**Comment by appetizermonster** [comment]
@notatestuser Sorry for late, I'm so busy for my work these days, I think it will be released with several another additions in at least two weeks.

**Comment by notatestuser** [comment]
Bump. I would really like to see this fixed. :smiley:
